### PR TITLE
[CI] Add ci workflows

### PR DESCRIPTION
**Pull request by red-0ne**
# CI Update

This pull request includes the following updates to the CI workflow:

## Issue and PR templates

Added the same issue and PR templates as in `poktroll`.

## Workflows

Added reviewdog CI workflow including
* `TODO_IN_THIS_` checks.
* Standard interface implementation checks.
* Spell checking
* Test runner

## Makefile targets

Added basic `Makefile` targets:
* `make test_all`
* `make go_lint`
* `TODO` related targets

### CODE REVIEW

**Review by okdas — APPROVED** [review]
Thank you @red-0ne!

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.22.2"
+
+      - name: Run golangci-lint
```
Comment (okdas) [inline .github/workflows/run-tests.yml:29]: We need to install and configure it first. I don't remember exactly why, but we don't use the [official github action](https://golangci-lint.run/welcome/install/#github-actions) on our main repo.

If you don't want to look into this now — I'm happy to pick it up! In that case please remove `make go_lint` step.

**Review by okdas — CHANGES_REQUESTED** [review]

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (red-0ne) [reply .github/workflows/run-tests.yml:29]: @okdas , I'd love to have a linter in this repo. Not sure how to proceed though.

Comment (okdas) [reply .github/workflows/run-tests.yml:29]: @red-0ne you can comment out this step, and I'll set up the `golangci-lint` next week then, would that work?

**Review by okdas — APPROVED** [review]
LGTM!

@red-0ne I've added a commit to remove a dependency on `check_go_version` make target. If this looks good to you, please go ahead and merge at your convenience.

**Review by Olshansk — APPROVED** [review]

**Pull request merged**